util/sqlexec/utils.go

```diff
@@ -135,86 +250,20 @@ func escapeSQL(sql string, args ...interface{}) ([]byte, error) {
 			if arg == nil {
 				buf = append(buf, "NULL"...)
 			} else {
-				switch v := arg.(type) {
-				case int:
-					buf = strconv.AppendInt(buf, int64(v), 10)
-				case int8:
-					buf = strconv.AppendInt(buf, int64(v), 10)
-				case int16:
-					buf = strconv.AppendInt(buf, int64(v), 10)
-				case int32:
-					buf = strconv.AppendInt(buf, int64(v), 10)
-				case int64:
-					buf = strconv.AppendInt(buf, v, 10)
-				case uint:
-					buf = strconv.AppendUint(buf, uint64(v), 10)
-				case uint8:
-					buf = strconv.AppendUint(buf, uint64(v), 10)
-				case uint16:
-					buf = strconv.AppendUint(buf, uint64(v), 10)
-				case uint32:
-					buf = strconv.AppendUint(buf, uint64(v), 10)
-				case uint64:
-					buf = strconv.AppendUint(buf, v, 10)
-				case float32:
-					buf = strconv.AppendFloat(buf, float64(v), 'g', -1, 32)
-				case float64:
-					buf = strconv.AppendFloat(buf, v, 'g', -1, 64)
-				case bool:
-					if v {
-						buf = append(buf, '1')
-					} else {
-						buf = append(buf, '0')
-					}
-				case time.Time:
-					if v.IsZero() {
-						buf = append(buf, "'0000-00-00'"...)
-					} else {
-						buf = append(buf, '\'')
-						buf = v.AppendFormat(buf, "2006-01-02 15:04:05.999999")
-						buf = append(buf, '\'')
-					}
-				case json.RawMessage:
-					buf = append(buf, '\'')
-					buf = escapeBytesBackslash(buf, v)
-					buf = append(buf, '\'')
-				case []byte:
-					if v == nil {
-						buf = append(buf, "NULL"...)
-					} else {
-						buf = append(buf, "_binary'"...)
-						buf = escapeBytesBackslash(buf, v)
-						buf = append(buf, '\'')
-					}
-				case string:
-					buf = append(buf, '\'')
-					buf = escapeStringBackslash(buf, v)
-					buf = append(buf, '\'')
-				case []string:
-					for i, k := range v {
-						if i > 0 {
-							buf = append(buf, ',')
+				if !handleDynTpArg(arg) {
+					reflectTp := reflect.TypeOf(arg)
```

Comment:
Try `switch reflect.TypeOf(arg).Kind()`. You don't need to `Convert()`.

Reply:
I'm not sure about your suggestion, how could the type be changed without `Convert`? And for `Struct` kind I think we must use other function to tell if they are same

https://github.com/golang/go/blob/f57f02fcd519a86683c47a444e0e24a086fea8e0/src/reflect/type.go#L1681

And I think I can improve current code for basic types by `switch kind` to avoid looping one-by-one to check ConvertibleTo

Reply:
> I know they have same _kind_, but what do you suggest to change the _type_ without `Convert`? https://go.dev/play/p/jw9FCSekc2s

You don't need to change the reflection type. Check https://go.dev/play/p/oh00SrayMmq.

Reply:
updated, I will try to handle customized types whose underlying type is slice later. And I noticed now the performance is a bit lower, will run benchmark later as well.

Reply:
Because you always call `reflect.TypeOf`. We could do  same as before, first we match by `v.(type)`, then we try `reflect.TypeOf`.

Reply:
fixed, and updated the benchmark in PR description. PTAL